### PR TITLE
add yar library  and RdKafka stubs

### PR DESCRIPTION
**Pull request by weizhimiao**

### CODE REVIEW

File: yar/yar.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+ * Class Yar_Server_Exception
+ */
+class Yar_Server_Exception extends Exception {
+    /* 属性 */
+    protected $_type ;
+
+    /**
+     * Retrieve exception's type
+     * Get the exception original type threw by server
+     * @return $type string
```
Comment (andrey-sokolov) [inline yar/yar.php]: should be just `@return string`

File: yar/yar.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+     *  If this callback is set, then Yar will call this callback while error occurred.
+     * @return $bool boolean
+     * @link http://php.net/manual/en/yar-concurrent-client.loop.php
+     */
+    public static function loop ($callback , $error_callback) {}
+
+    /**
+     * Clean all registered calls
+     * Clean all registered calls
+     * @return $bool bool
```
Comment (andrey-sokolov) [inline yar/yar.php]: should be `@return bool`

File: yar/yar.php [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    public static function call (  $uri ,  $method ,  $parameters, ...$callback ){}
+
+    /**
+     * Send all calls
+     * @param $callback
+     *  If this callback is set, then Yar will call this callback after all calls are sent and before any response return, with a $callinfo NULL.
+     *  Then, if user didn't specify callback when registering concurrent call, this callback will be used to handle response, otherwise, the callback specified while registering will be used.
+     * @param $error_callback
+     *  If this callback is set, then Yar will call this callback while error occurred.
+     * @return $bool boolean
```
Comment (andrey-sokolov) [inline yar/yar.php]: should be `@return boolean`

File: yar/yar.php [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    static $_callback ;
+    static $_error_callback ;
+
+    /**
+     * Register a concurrent call
+     * @param $uri string The RPC server URI(http, tcp)
+     * @param $method string Service name(aka the method name)
+     * @param $parameters array Parameters
+     * @param array ...$callback A function callback, which will be called while the response return.
+     * @return $id int An unique id, can be used to identified which call it is.
```
Comment (andrey-sokolov) [inline yar/yar.php]: Please, replace  with `@return int`

File: yar/yar.php [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+     * @param $parameters array Parameters.
+     * @link http://php.net/manual/en/yar-client.call.php
+     */
+    public function __call ( $method , $parameters ){}
+
+    /**
+     * Create a client
+     * Yar_Client constructor.
+     * @param $url string Yar Server URL.
+     * @return $instance object Yar_Client instance.
```
Comment (andrey-sokolov) [inline yar/yar.php]: `@return` is redundant in constructor

File: yar/yar.php [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    /* 属性 */
+    protected $_executor ;
+
+    /**
+     * Register a server
+     * Set up a Yar HTTP RPC Server, All the public methods of $obj will be register as a RPC service.
+     *
+     * Yar_Server constructor.
+     * @param $obj object An Object, all public methods of its will be registered as RPC services.
+     * @return object An instance of Yar_Server.
```
Comment (andrey-sokolov) [inline yar/yar.php]: return is redundant in __construct

**Comment by EdwinHoksberg** [comment]
Hi, I'm also interested in adding the php-rdkafka extension stubs, the documentation for this extension can be found here: https://arnaud-lb.github.io/php-rdkafka/phpdoc/book.rdkafka.html

I don't know about the yar extension.

**Comment by MaXal** [comment]
@weizhimiao Sorry about closing/reopening but this is the only way to retrigger Travis build and we improve our test coverage based on your request.

**Comment by weizhimiao** [comment]
doc links： http://php.net/manual/en/book.yar.php

**Pull request merged**